Travis for install script

smits opened the pull request.

Commit:
Add travis script to test installation script

Commit:
Silly update to trigger travis

Commit:
Remove comments

Commit:
Fix travis file extension

Commit:
Fix apt dependency

Commit:
Change prefix to a writable place and enable all features

Commit:
Remove compiler matrix as it doesn't work properly

meyerj committed:
Merge branch 'toolchain-2.9' into smits-patch-1

Commit:
Only use containerized builds

meyerj committed:
Fix readlink -f on MacOS X

meyerj committed:
install.sh: do not call lsb_release on non-Linux platforms

Commit:
Merge pull request #16 from orocos-toolchain/smits-patch-1

Preliminary merge

Commit:
Add missing dependency if reinstall if required

Signed-off-by: Ruben Smits <ruben@intermodalics.eu>

Commit:
Make travis build verbose

Signed-off-by: Ruben Smits <ruben@intermodalics.eu>

Commit:
replace . with /Users/rubensmits/orocos-toolchain for OSX

Signed-off-by: Ruben Smits <ruben@intermodalics.eu>

Commit:
Update RTT

meyerj committed:
install.sh: fixed type and verbose display of features

Commit:
Link gnu-getopt

Signed-off-by: Ruben Smits <ruben@intermodalics.eu>

Commit:
Force linking of gnu-getopt

Signed-off-by: Ruben Smits <ruben@intermodalics.eu>

Commit:
Add apt addons to matrix entry

Signed-off-by: Ruben Smits <ruben@intermodalics.eu>

meyerj committed:
rtt: updated submodule to include Boost 1.62 patches

meyerj committed:
rtt: updated submodule

Commit:
Remove broken xenial build matrix entry

Signed-off-by: Ruben Smits <ruben@intermodalics.eu>

Commit:
Update rtt

Signed-off-by: Ruben Smits <ruben@intermodalics.eu>

meyerj committed:
rtt: updated submodule

meyerj committed:
Merge branch 'installation-script' into travis-for-install-script

meyerj committed:
update submodule rtt

meyerj committed:
updated submodules rtt and ocl

meyerj commented on .travis.yml:
```diff
@@ -0,0 +1,33 @@
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew link --force gnu-getopt ; fi
+
+script:
+  - ./install.sh --verbose --prefix $HOME/install --enable-tests --enable-corba --enable-mqueue
```
We don't actually run the tests, so why build them? Or should we add another line that runs the tests after installation? Or would you expect that `--enable-tests` not only builds but also runs the tests?

Commit:
travis: add readline dependency for OCL

Signed-off-by: Ruben Smits <ruben@intermodalics.eu>

meyerj committed:
travis: disable building of tests

Unit tests will be built and run for individual projects, but not for this meta repository. The purpose of this job is mainly to test the installation script.

meyerj commented:
Looks good to merge now. Note that containerized builds still cause some trouble for unit testing, like in https://github.com/orocos-toolchain/rtt/pull/195, but as a pure installation script test case it should be fine. I disabled tests in 71d9b01, but they were only built and not executed until now.
